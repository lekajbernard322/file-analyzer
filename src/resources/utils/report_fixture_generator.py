from os import listdir
from os.path import isfile, join
import json
from datetime import datetime, timezone
from bson.objectid import ObjectId

INPUT_DIR = r'../report_data/'
OUTPUT_PATH = r'../../backend/app/report/fixtures/report_fixture.json'

file_paths = [(join(INPUT_DIR, file_name), file_name)
              for file_name in listdir(INPUT_DIR)
              if isfile(join(INPUT_DIR, file_name))]


def prepare_output_data(i, file_name, content):
    return {
        'model': 'report.report',
        'pk': str(ObjectId()),
        'fields': {
            'name': file_name,
            'content': content,
            'created': datetime.now(tz=timezone.utc).isoformat()
        }
    }


data = []
for index, (input_file_path, input_file_name) in enumerate(file_paths):
    print(f'[*] Processing file: {input_file_path}')

    with open(input_file_path, 'r') as input_file:
        json_content = json.load(input_file)

        output_data = prepare_output_data(index, input_file_name, json_content)

        data.append(output_data)

print(f'[*] Dumping to fixture file: {OUTPUT_PATH}')
with open(OUTPUT_PATH, 'w', encoding='utf-8') as output_file:
    json.dump(data, output_file, indent=4)

print(f'[\u2714] Done')
