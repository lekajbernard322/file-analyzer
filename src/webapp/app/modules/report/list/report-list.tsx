import React, {useEffect, useState} from 'react';
import {useAppDispatch, useAppSelector} from "../../../configuration/store";
import {getDllStatistics, getFunctionStatistics, getReports, clearReportList} from "./report-list.reducer";
import {Headline, Subheading} from "../../common/typography";
import HorizontalRule from "../../common/horizontal-rule";
import Breadcrumbs from "../../common/breadcrumbs";
import InformationHighlighter from "../../common/information-highlighter";
import HorizontalContainer from "../../common/horizontal-container";
import PageContent from "../../common/page-content";
import SearchField from "../../common/search-field";
import ReportTable from "./report-table";
import {Trans, useTranslation} from "react-i18next";

const ReportListPage = () => {
    const dispatch = useAppDispatch();
    const reports = useAppSelector(store => store.report.list.data);
    const totalNumberOfReports = useAppSelector(store => store.report.list.total);
    const functionStatistics = useAppSelector(store => store.report.list.functionStatistics);
    const dllStatistics = useAppSelector(store => store.report.list.dllStatistics);

    const {t} = useTranslation();

    const [searchTerm, setSearchTerm] = useState('');

    useEffect(() => {
        searchReports(searchTerm);
        dispatch(getFunctionStatistics());
        dispatch(getDllStatistics());
    }, []);

    const searchReportsInner = (search, offset = 0, limit = 2) => {
        dispatch(getReports({search, limit, offset}));
    }

    const searchReports = (search, offset = 0, limit = 2) => {
        dispatch(clearReportList())
        searchReportsInner(search, offset, limit);
    }

    const loadMoreReports = () => {
        searchReportsInner(searchTerm, reports?.length);
    }

    return (
        <div>
            <Breadcrumbs config={[{label: t('breadcrumb.reports'), to: '/'}]}/>
            <Headline message={t('reports.title')}/>
            <Subheading message={t('reports.subtitle')}/>

            <HorizontalRule/>

            <HorizontalContainer>
                <InformationHighlighter message={t('reports.totalNumberOfReports')}
                                        highlightedInformation={totalNumberOfReports}/>
                <InformationHighlighter message={t('reports.averageNumberOfDlls')}
                                        highlightedInformation={functionStatistics?.averageNumberOfFunctions}/>
                <InformationHighlighter message={t('reports.averageNumberOfFunctions')}
                                        highlightedInformation={dllStatistics?.averageNumberOfDlls}/>
            </HorizontalContainer>

            <HorizontalRule/>

            <SearchField searchTerm={searchTerm} onChangeSearchTerm={setSearchTerm}
                         onPressEnter={() => searchReports(searchTerm)}/>
            <button onClick={() => searchReports(searchTerm)}><Trans i18nKey={'common.filter'}/></button>

            <PageContent>
                <ReportTable reports={reports} disabled={reports?.length === totalNumberOfReports}
                             onLoadMore={loadMoreReports}/>
            </PageContent>
        </div>
    );
};

export default ReportListPage;
