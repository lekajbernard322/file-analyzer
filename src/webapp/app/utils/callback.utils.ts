export const createOnPressEnter = (callback) => ({key}) => key === 'Enter' && callback && callback()
