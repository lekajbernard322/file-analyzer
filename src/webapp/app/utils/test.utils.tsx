import thunk from "redux-thunk";
import {createMemoryHistory} from "history";
import {render} from "@testing-library/react";
import {Provider} from "react-redux";
import {Router} from "react-router-dom";
import React from "react";
import configureStore from "redux-mock-store";

export const renderWithContext = (additionalState, ui) => {
    const mockStore = configureStore([thunk]);
    const store = mockStore(additionalState);
    const history = createMemoryHistory();

    const Wrapper = ({children}) => (
        <Provider store={store}>
            <Router history={history}>
                {children}
            </Router>
        </Provider>);

    return {rendered: render(ui, {wrapper: Wrapper}), store, history};
};
