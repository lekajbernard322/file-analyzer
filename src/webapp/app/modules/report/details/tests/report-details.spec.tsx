import React from "react";
import ReportDetailsPage from "../report-details";
import {renderWithContext} from "../../../../utils/test.utils";
import {
    getReportDetails,
    getDllOccurrences,
} from "../report-details.reducer";
import {getDllStatistics, getFunctionStatistics} from "../../list/report-list.reducer";

export const mockData = {
    report: {
        details: {
            data: {
                _id: "1",
                name: "report1",
                created: "some date",
                content: [
                    {
                        DLLName: "test.dll",
                        Functions: ["function1", "function2"],
                    },
                ],
            },
            dllOccurrences: {
                "test.dll": 3,
            },
        },
        list: {
            functionStatistics: {
                averageNumberOfFunctions: 1,
                minNumberOfFunctions: 2,
                maxNumberOfFunctions: 3,
            },
            dllStatistics: {
                averageNumberOfDlls: 4,
                minNumberOfDlls: 5,
                maxNumberOfDlls: 6,
            },
        },
    },
};

describe("ReportDetail", () => {
    const renderComponent = () => {
        return renderWithContext(mockData, <ReportDetailsPage/>);
    };

    it("should render correctly", () => {
        const {
            rendered: {container},
        } = renderComponent();

        expect(container).toMatchSnapshot();
    });

    it("should execute fetches for data", () => {
        const {store} = renderComponent();

        const [
            getFunctionStatisticsPendingAction,
            getDllStatisticsPendingAction,
            getReportDetailsPendingAction,
            getDllOccurrencesPendingAction
        ] = store.getActions();

        expect(getFunctionStatisticsPendingAction).toMatchObject({
            type: getFunctionStatistics.pending.type,
        });
        expect(getDllStatisticsPendingAction).toMatchObject({
            type: getDllStatistics.pending.type,
        });
        expect(getReportDetailsPendingAction).toMatchObject({
            type: getReportDetails.pending.type,
        });
        expect(getDllOccurrencesPendingAction).toMatchObject({
            type: getDllOccurrences.pending.type,
        });
    });
});
