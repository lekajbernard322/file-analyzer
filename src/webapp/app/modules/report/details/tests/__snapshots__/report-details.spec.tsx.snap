// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportDetail should render correctly 1`] = `
<div>
  <div>
    <div
      class="breadcrumb"
    >
      <a
        href="/"
      >
        Reports
        /
      </a>
      <a
        href="/"
      >
        Details
        /
      </a>
    </div>
    <h1
      class="headline"
    >
      report1
    </h1>
    <h3
      class="subheading"
    >
      Details of PE metadata showing the list of used DLLs and their functions
    </h3>
    <div>
      <hr />
    </div>
    <div
      class="horizontal-container"
    >
      <div
        class="information-highlighter"
      >
        <div
          class="information-highlighter-message"
        >
          Number of DLLs, which is lower than the average [4]
        </div>
        <div
          class="information-highlighter-value"
        >
          1
        </div>
      </div>
      <div
        class="information-highlighter"
      >
        <div
          class="information-highlighter-message"
        >
          Number of functions, which is higher than the average [1]
        </div>
        <div
          class="information-highlighter-value"
        >
          2
        </div>
      </div>
    </div>
    <div>
      <hr />
    </div>
    <input
      placeholder="Search"
      value=""
    />
    <div
      class="page-content"
    >
      <div>
        <ul
          class="report-details"
        >
          <li
            class="dll-name"
          >
            test.dll
             
            <span>
              (
              occurrences in other reports
              : 
              3
              )
            </span>
            <ul>
              <li
                class="function-name"
              >
                function1
              </li>
              <li
                class="function-name"
              >
                function2
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
